sys/suit/v4/cbor.c

```diff
@@ -40,159 +40,142 @@
 static suit_manifest_handler_t _manifest_get_auth_wrapper_handler(int key);
 typedef suit_manifest_handler_t (*suit_manifest_handler_getter_t)(int key);
 
-int suit_cbor_map_iterate_init(CborValue *map, CborValue *it)
+int suit_cbor_map_iterate_init(nanocbor_value_t *map, nanocbor_value_t *it)
 {
-    if (!cbor_value_is_map(map)) {
+    if (nanocbor_get_type(map) != NANOCBOR_TYPE_MAP) {
         LOG_INFO("suit_v4_parse(): manifest not a map\n)");
         return SUIT_ERR_INVALID_MANIFEST;
     }
 
-    cbor_value_enter_container(map, it);
+    nanocbor_enter_map(map, it);
 
     return SUIT_OK;
 }
 
-int suit_cbor_map_iterate(CborValue *it, CborValue *key, CborValue *value)
+int suit_cbor_map_iterate(nanocbor_value_t *it, nanocbor_value_t *key,
+                          nanocbor_value_t *value)
 {
-    if (cbor_value_at_end(it)) {
+    if (nanocbor_at_end(it)) {
         return 0;
     }
 
     *key = *it;
-    cbor_value_advance(it);
+    nanocbor_skip(it);
 
     *value = *it;
-    cbor_value_advance(it);
+    nanocbor_skip(it);
 
     return 1;
 }
 
-int suit_cbor_get_int(const CborValue *it, int *out)
+int suit_cbor_get_int32(nanocbor_value_t *it, int32_t *out)
 {
-    if (!cbor_value_is_integer(it)) {
-        LOG_DEBUG("expected integer type, got %u\n", cbor_value_get_type(it));
-        return SUIT_ERR_INVALID_MANIFEST;
-    }
+    int res = nanocbor_get_int32(it, out);
 
-    /* This check tests whether the integer fits into "int", thus the check
-     * is platform dependent. This is for lack of specification of actually
-     * allowed values, to be made explicit at some point. */
-    if (cbor_value_get_int_checked(it, out) == CborErrorDataTooLarge) {
-        LOG_DEBUG("integer doesn't fit into int type\n");
+    if (res < NANOCBOR_OK) {
+        LOG_DEBUG("suit_cbor_get_int32() error %u\n", res);
         return SUIT_ERR_INVALID_MANIFEST;
     }
 
     return SUIT_OK;
 }
 
-int suit_cbor_get_string(const CborValue *it, const uint8_t **buf, size_t *len)
+int suit_cbor_get_string(nanocbor_value_t *it, const uint8_t **buf, size_t *len)
 {
-    if (!(cbor_value_is_text_string(it) || cbor_value_is_byte_string(it) || cbor_value_is_length_known(it))) {
+    // TODO: there was a check for len before
```

Comment:
doxygen style comments?

Reply:
I removed the comment. I think it is not necessary anymore.

Reply:
True not using the length anymore.